tfhe/src/strings/server_key/mod.rs

```diff
@@ -7,12 +7,39 @@ pub use trim::split_ascii_whitespace;
 
 use crate::integer::bigint::static_unsigned::StaticUnsignedBigInt;
 use crate::integer::prelude::*;
-use crate::integer::{BooleanBlock, RadixCiphertext, ServerKey};
+use crate::integer::{BooleanBlock, RadixCiphertext, ServerKey as IntegerServerKey};
 use crate::strings::ciphertext::{num_ascii_blocks, FheAsciiChar, FheString};
 use crate::strings::N;
 use rayon::prelude::*;
+use std::borrow::Borrow;
 use std::cmp::Ordering;
 
+pub struct ServerKey<T>
+where
+    T: Borrow<IntegerServerKey> + Sync,
```

Comment:
Is the `Sync` bound really needed ?

as most if not all function would first call `let intsk = self.inner.borrow()` 
and the int sk is the one that needs to be Sync, and it is

Reply:
`compare_shifted_index` needs it for example

We could borrow outside of the parallel iterator and rebuild a `strings::ServerKey<&ServerKey>` in the closure to be able to remove this bound.
Do you think it would be better this way?

Reply:
Hum I think I would remove the bound from the struct type, so that the Sync only appears where it is actually needed

Reply:
Seems simpler to me to have a single global constraint even if this is a bit more restrictive

Reply:
Yeah we can do that